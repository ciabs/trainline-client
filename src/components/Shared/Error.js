import React from 'react';
import FontAwesome  from 'react-fontawesome';

export const Error = () => (
  <FontAwesome
    style={{color: 'red', position: 'absolute', top: '15px', right: '15px'}}
    name='exclamation-triangle'
    size='2x'
  />
);
