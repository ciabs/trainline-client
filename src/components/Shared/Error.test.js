import React from 'react';
import renderer from 'react-test-renderer';
import {Error} from './Error';
import ReactDOM from 'react-dom';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('Error', () => {
  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Error />,
      div
    );
  });

  it('should render without crashing when mounted with Enzyme', () => {
    shallow(<Error />);
  });

  it('should should have a valid snapshot', () => {
    const component = renderer.create(
      <Error />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
