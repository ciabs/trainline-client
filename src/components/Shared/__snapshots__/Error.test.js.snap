// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should should have a valid snapshot 1`] = `
<span
  aria-hidden={true}
  className="fa fa-exclamation-triangle fa-2x"
  style={
    Object {
      "color": "red",
      "position": "absolute",
      "right": "15px",
      "top": "15px",
    }
  }
/>
`;
